packages/edit-site/src/components/layout/router.js

```diff
@@ -17,38 +17,89 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
 import SidebarNavigationScreenGlobalStyles from '../sidebar-navigation-screen-global-styles';
 import SidebarNavigationScreenMain from '../sidebar-navigation-screen-main';
 import SidebarNavigationScreenNavigationMenus from '../sidebar-navigation-screen-navigation-menus';
-import SidebarNavigationScreenPage from '../sidebar-navigation-screen-page';
 import SidebarNavigationScreenTemplatesBrowse from '../sidebar-navigation-screen-templates-browse';
-import SidebarNavigationScreenTemplate from '../sidebar-navigation-screen-template';
-import SidebarNavigationScreenPattern from '../sidebar-navigation-screen-pattern';
 import SidebarNavigationScreenPatterns from '../sidebar-navigation-screen-patterns';
 import SidebarNavigationScreenNavigationMenu from '../sidebar-navigation-screen-navigation-menu';
 import DataViewsSidebarContent from '../sidebar-dataviews';
+import {
+	NAVIGATION_POST_TYPE,
+	PATTERN_TYPES,
+	TEMPLATE_PART_POST_TYPE,
+	TEMPLATE_POST_TYPE,
+} from '../../utils/constants';
 
 const { useLocation, useHistory } = unlock( routerPrivateApis );
 
-export default function useLayoutAreas() {
-	const isSiteEditorLoading = useIsSiteEditorLoading();
+function useRedirectOldPaths() {
 	const history = useHistory();
 	const { params } = useLocation();
-	const { postType, postId, path, layout, isCustom, canvas } = params;
-
 	useEffect( () => {
-		// `/wp_template_part/all` path is no longer used and redirects to
-		// Patterns page for backward compatibility.
+		const {
+			postType,
+			postId,
+			path,
+			layout,
+			isCustom,
+			canvas,
+			categoryType,
+			...rest
+		} = params;
+
 		if ( path === '/wp_template_part/all' ) {
-			history.replace( { path: '/patterns' } );
+			history.replace( { postType: TEMPLATE_PART_POST_TYPE } );
+		}
+
+		const allParamsButPath = {
+			postId,
+			layout,
+			isCustom,
+			canvas,
+			...rest,
+		};
+
+		if ( path === '/page' ) {
+			history.replace( {
+				postType: 'page',
+				...allParamsButPath,
+			} );
 		}
-	}, [ history, path ] );
 
-	// Note: Since "sidebar" is not yet supported here,
-	// returning undefined from "mobile" means show the sidebar.
+		if ( path === '/wp_template' ) {
+			history.replace( {
+				postType: TEMPLATE_POST_TYPE,
+				...allParamsButPath,
+			} );
+		}
+
+		if ( path === '/patterns' ) {
+			history.replace( {
+				postType: categoryType ?? PATTERN_TYPES.theme,
+				...allParamsButPath,
+			} );
+		}
+
+		if ( path === '/navigation' ) {
+			history.replace( {
+				postType: NAVIGATION_POST_TYPE,
+				...allParamsButPath,
+			} );
+		}
+	}, [ history, params ] );
+}
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const history = useHistory();
+	const { params } = useLocation();
+	const { postType, postId, path, layout, isCustom, canvas, categoryType } =
+		params;
+	useRedirectOldPaths();
 
 	// Page list
-	if ( path === '/page' ) {
+	if ( postType === 'page' ) {
```

Comment:
Seems the only `path` remaining is `/wp_global_styles`, right? It should be the first `if` in this function. If I go to an ambiguous URL like `path=/wp_global_styles&postType=wp_template`, the `path` should win.

Also, one gap in the current UI is what kind of sidebar to show for post types that are not explicitly handled in this function. If I go to `postType=post&postId=42`, it will show the right post (because that's handled by `useInitEditedEntityFromUrl`), but will show the root sidebar. This problem is not new though, the old version would show you the Navigations sidebar, also not great.

Reply:
By the way, if we want to forbid editing posts and other post types in the site editor, what is the idea behind supporting page editing there? What is so special about pages that we need a second UI for them in addition to `edit.php?post_type=page`?

Reply:
There's nothing special about it. Maybe they're a bit closer to "design/appearance" than the rest of post types. But ultimately, they don't belong there, it's just a tactical move as we iterate towards the redesign of WP-Admin. More details here https://github.com/WordPress/gutenberg/issues/55238#issuecomment-2118253846

Reply:
Thanks, that explains a lot 🙂

Reply:
> the only path remaining is `/wp_global_styles`, right? It should be the first `if` in this function.

This feedback is I believe still valid, but unaddressed. 🙂